Build with flag(disable-git-info) fails

I was building a [statically linked aarch64 stack binary](https://github.com/bjin/stack-aarch64-static/) for use with aarch64 alpine docker, since I was grabbing tarball file from releases page instead of `git clone`, I thought it might be better to disable git hash to avoid confusing. But I encountered following compilation error:

```
/home/circleci/project/stack-2.11.1/src/main/BuildInfo.hs:48:18: error:
    Variable not in scope: showStackVersion :: [Char]
   |
48 | versionString' = showStackVersion ++ afterVersion
   |                  ^^^^^^^^^^^^^^^^
```

It seems that the offending commit is https://github.com/commercialhaskell/stack/commit/43b1210a5316f5e1131c3be0cb6989290ffbef8d. The import list of `src/main/BuildInfo.hs` is not changed accordingly in `USE_GIT_INFO` guard.

--- a/.github/workflows/unit-tests.yml
+++ b/.github/workflows/unit-tests.yml
@@ -9,7 +9,7 @@ on:
     - rc/**
   workflow_dispatch:
 
-# As of 5 May 2023, ubuntu-latest, windows-latest and macos-latest come
+# As of 22 May 2023, ubuntu-latest, windows-latest and macos-latest come
 # with Stack 2.9.3 and GHC 9.6.1.
 
 jobs:
@@ -28,9 +28,10 @@ jobs:
           ${{ runner.os }}-
     - name: Pedantic build
       run: |
-        # Stack 2.9.3 is required to build Stack
-        stack upgrade
         stack build --pedantic
+    - name: disable-git-info build
+      run: |
+        stack build --pedantic --flag stack:disable-git-info
   unit-tests:
     name: Unit tests
     runs-on: ${{ matrix.os }}

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -14,6 +14,9 @@ Other enhancements:
 
 Bug fixes:
 
+* Restore building of Stack with Cabal flag `disable-git-info` (broken with
+  Stack 2.11.1).
+
 ## v2.11.1 - 2023-05-18
 
 **Changes since v2.9.3:**

--- a/src/main/BuildInfo.hs
+++ b/src/main/BuildInfo.hs
@@ -16,21 +16,18 @@ module BuildInfo
 #ifndef HIDE_DEP_VERSIONS
 import qualified Build_stack
 #endif
-#ifdef USE_GIT_INFO
 import           Data.Version ( versionBranch )
-#else
-import           Data.Version ( showVersion, versionBranch )
-#endif
 import           Distribution.System ( buildArch )
 import qualified Distribution.Text as Cabal ( display )
 #ifdef USE_GIT_INFO
 import           GitHash ( giCommitCount, tGitInfoCwdTry )
-#endif
-#ifdef USE_GIT_INFO
 import           Options.Applicative.Simple ( simpleVersion )
 #endif
-import           Stack.Prelude
 import qualified Paths_stack as Meta
+import           Stack.Prelude
+#ifndef USE_GIT_INFO
+import           Stack.Types.Version ( showStackVersion )
+#endif
 
 versionString' :: String
 #ifdef USE_GIT_INFO

